pkg/controller/storagecluster/external_resources.go

```diff
@@ -123,25 +143,38 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterResources(ownerRef
 		case "StorageClass":
 			var sc *storagev1.StorageClass
 			if d.Name == cephFsStorageClassName {
-				// Setting the fsname for cephfs StorageClass
+				// Setting the 'sc' point to CephFS StorageClass
 				sc = scs[0]
+				enableRookCSICephFS = true
 			} else if d.Name == cephRbdStorageClassName {
-				// Setting the PoolName for RBD StorageClass
+				// Setting the 'sc' point to RBD StorageClass
 				sc = scs[1]
 			} else if d.Name == cephRgwStorageClassName {
 				// Set the external rgw endpoint variable for later use on the Noobaa CR (as a label)
 				// Replace the colon with an underscore, otherwise the label will be invalid
 				externalRgwEndpointReplaceColon := strings.Replace(d.Data[externalCephRgwEndpointKey], ":", "_", -1)
 				externalRgwEndpoint = externalRgwEndpointReplaceColon
 
-				// Setting the Endpoint for OBC StorageClass
+				// Setting the 'sc' point to OBC StorageClass
 				sc = scs[2]
 			}
 			for k, v := range d.Data {
 				sc.Parameters[k] = v
 			}
+			availableSCs = append(availableSCs, sc)
 		}
 	}
+	// creating only the available storageClasses
+	err = r.createStorageClasses(availableSCs, reqLogger)
+	if err != nil {
+		reqLogger.Error(err, "failed to create needed StorageClasses")
+		return err
+	}
+	if err = r.setRookCSICephFS(enableRookCSICephFS, instance, reqLogger); err != nil {
```

Comment:
Here, we could have added a `if` condition first, to check something like,
```suggestion
	if !enableRookCSICephFS {
		if err = r.setRookCSICephFS(enableRookCSICephFS, instance, reqLogger); err != nil {
```
That is, when `CephFS` storageclass is not enabled, then only call the `setRookCSICephFS()` to disable the CSI drivers.
My reasoning to not to call it that way is, on a later stage when we decide to `reconcile` external resource creation, the above part of the code don't have to be changed.
Let me know what others think?

Reply:
It took me a bit to parse, but I think I understand. The point is you always want to set the CephFS CSI config to *something*, whether true or false. This is good!

Reply:
I agree this should be caught in the function, not outside.